### DISCUSSION TITLE
Regarding Diofant's solver

### DISCUSSION THREAD

**Discussion by anutosh491** [diofant/diofant#1186] [Q&A] [answered]
Hello @skirpichev , I wanted to know whether there is any way to solve something like `x = sin(x)` or `x = log(x)` ,`x = exp(x)` , `x = exp(1/x) `. First one should return 0 in the Real domain and the next 2 should return an empty set . Sympy's` solveset()` method returns something like `ConditionSet(x, Eq(x - sin(x), 0), Complexes), ConditionSet(x, Eq(x - exp(x), 0), Complexes)` which is not wrong but not what I would like and Sympy's solvify returns Not implemented for these !

**Comment by skirpichev** [comment, marked as answer]
> x = sin(x) or x = log(x) ,x = exp(x) , x = exp(1/x) 

First can't be solved with known special functions (AFAIK).  Others - can be solved with the LambertW.

**Reply by anutosh491** [reply]
Thanks You for the help ! If we break sin(x) as `-I*(exp(I*x) - exp(-I*x))/2` even this could be answered using LambertW

**Reply by skirpichev** [reply]
@anutosh491, I doubt.  Not with the standard LambertW.   But, probably, with its generalization, see [this](https://en.wikipedia.org/wiki/Lambert_W_function#Generalizations).
